### PR TITLE
Localize date and time.

### PR DESCRIPTION
**Pull request by dzatoah**

### CODE REVIEW

**Comment by dzatoah** [comment]
Mh it's 11:48 here in germany, and it shows `11/15/2023 20:55:46`...

**Review by sunweaver — DISMISSED** [review]
LGTM

**Comment by dzatoah** [comment]
We don't need any JS: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local
(Thanks to our costumer Uni-Bonn for the idea)

**Comment by gber** [comment]
> We don't need any JS: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local (Thanks to our costumer Uni-Bonn for the idea)

Yes, please convert the date/time from privacyIDEA and pass it to Smarty using the format described here: https://developer.mozilla.org/en-US/docs/Web/HTML/Date_and_time_formats#local_date_and_time_strings.

File: personal/privacyidea/default/mfa_intro.tpl [hunk 1]
```diff
@@ -237,7 +244,16 @@
                         <td><button class="txtonlybtn" type="submit" name="mfaTokenAction[mfaTokenView]" value="{$token.serial}">
                             {if strpos($tokenDescriptionACL, "r") !== false}{$token.description}{else}{t}not shown{/t}{/if}
                         </button></td>
-                        <td>{if strpos($tokenLastUsedACL, "r") !== false}{$token.info.last_auth}{else}{t}not shown{/t}{/if}</td>
+
+                        {* token.info.last_auth *}
+                        {if !empty($token.info.last_auth)}
+                            <td class="tokenLastUsed">
+                                {if strpos($tokenLastUsedACL, "r") !== false}<input type="datetime-local" readOnly value="{$token.info.last_auth}">{else}{t}not shown{/t}{/if}
```
Comment (gber) [inline personal/privacyidea/default/mfa_intro.tpl:251]: `readonly` is included in a submit, better use `disabled`.

**Review by gber — CHANGES_REQUESTED** [review]

File: personal/privacyidea/default/mfa_intro.tpl (hunk 1 above)
Comment (dzatoah) [reply personal/privacyidea/default/mfa_intro.tpl:251]: What exactly do you mean `readonly` is included in a submit?

`disabled` makes the input look gray.

Comment (gber) [reply personal/privacyidea/default/mfa_intro.tpl:251]: Both `readonly` does not prevent interaction, i.e. the value cannot be changed but the element still receives focus which we do not want here. Furthermore the value is still submitted. See https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly#attribute_interactions for details.

Comment (gber) [reply personal/privacyidea/default/mfa_intro.tpl:251]: After some testing with different browsers it has become clear the `<input type=datetime-local>` is unsuitable here as it cannot be styled. Firefox e.g. shows a reset icon even in disabled state and takes too much whitespace.

**Comment by gber** [comment]
As `<input>` is unsuitable I have reverted back to the initial idea of using progressive enhancement with JS formatting of ISO 8601 dates baked into the templates.

**Comment by sunweaver** [comment]
Ack. Thanks for the research.

**Review by sunweaver — APPROVED** [review]
Well then. We'll got the JS route here, then.

**Pull request merged**